### PR TITLE
Added Multi-Select widget

### PR DESCRIPTION
**Pull request by tibrahulz**
## Describe your changes
- Added Multi Select widget as per requirement.

## Issue ticket number and link
https://github.com/EnsembleUI/ensemble-react/issues/39

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have added tests

### CODE REVIEW

File: packages/runtime/src/widgets/MultiSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  const handleChange = (value: string[]) => {
+    setSelectedValue(value)
+  };
+
+  return (
+    <SelectComponent
+      mode="multiple"
+      style={{ width: width || '100%' }}
+      placeholder={placeholder || 'Select'}
```
Comment (evshi) [inline packages/runtime/src/widgets/MultiSelect.tsx:48]: Need to resolve this binding.

File: packages/runtime/src/widgets/MultiSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { Expression } from 'framework';
+import { EnsembleWidgetProps } from '../util/types';
+
+type SelectOption = {
+  label: Expression<string>;
+  value: Expression<string | number>;
+}
+export type MultiSelectProps = {
+  placeholder?: Expression<string>;
+  data: SelectOption[];
```
Comment (evshi) [inline packages/runtime/src/widgets/MultiSelect.tsx:15]: Data must be a binding/expression

**Review by evshi — CHANGES_REQUESTED** [review]
You need to use the `useRegisterBindings` to binding the values to the ensemble state so they can be read later.

File: apps/starter/src/ensemble/collections.yaml [hunk 3]
```diff
@@ -17,3 +17,19 @@ View:
                     navigateModalScreen:
                       name: Ask Embrace
                       maskClosable: false
+        - Row:
+            mainAxis: spaceBetween
+            crossAxis: center
+            children:
+              - Markdown:
+                  text: "### Groups"
+              - MultiSelect:
```
Comment (evshi) [inline apps/starter/src/ensemble/collections.yaml:26]: Just sanity checking here, there's no existing widget in Ensemble that has this multi select functionality right?

File: packages/runtime/src/widgets/MultiSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { Expression, useTemplateData } from "framework";
+import { WidgetRegistry } from "../registry";
+import { PlusCircleOutlined } from "@ant-design/icons";
+import { Select as SelectComponent, Space } from "antd";
+import { get } from "lodash-es";
+import { SearchStyles } from "../util/types";
+import { getColor } from "../util/utils";
+
+interface EnsembleWidgetProps<T> {
```
Comment (evshi) [inline packages/runtime/src/widgets/MultiSelect.tsx:10]: Can you import this type from utils for now? Will refactor later to clean up

File: packages/runtime/src/widgets/MultiSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  placeholder,
+  styles,
+}) => {
+  const [options, setOptions] = useState<SelectOption[]>([]);
+  const [newOption, setNewOption] = useState("");
+  const [selectedValues, setSelectedValues] = useState<string[] | undefined>(
+    defaultOptions?.map((item) => item.value?.toString())
+  );
+
+  const templateData = useMemo(() => useTemplateData(data), [data]);
```
Comment (evshi) [inline packages/runtime/src/widgets/MultiSelect.tsx:43]: Shouldn't need useMemo here, useTemplateData already memoizes the value

**Review by evshi — APPROVED** [review]
Go ahead and merge if needed to unblock @sibteali786. We'll need probably need to do another pass at this like Search

File: apps/starter/src/ensemble/collections.yaml (hunk 3 above)
Comment (anserwaseem) [reply apps/starter/src/ensemble/collections.yaml:26]: yes, right

**Review by evshi — APPROVED** [review]

**Pull request merged**